application-tour-ui/src/main/resources/TourCode/TourJS.xml

```diff
@@ -257,15 +257,12 @@ require(['jquery', 'xwiki-meta'], function ($, xm) {
       });
       
       // Create the steps
-      var i, n     = jsonData.steps.length;
-      for (var i = 0; i &lt; n; i++) {
-        tour.addStep({
-          element   : jsonData.steps[i].element,
-          title     : jsonData.steps[i].title,
-          content   : jsonData.steps[i].content,
-          placement : jsonData.steps[i].placement,
-          backdrop  : jsonData.steps[i].backdrop
+      for (var i = 0; i &lt; jsonData.steps.length; i++) {
+        var step = {};
+        $.each(jsonData.steps[i], function(key, value) {
+          step[key] = value;
         });
+        tour.addStep(step);
```

Comment:
@gdelhumeau no objection, as long as you know that ``jsonData.steps`` is not being modified in some other place (that would trigger side effects on the tour).